frontend/src/metabase/components/DimensionLabel/DimensionLabel.jsx

```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import Dimension from "metabase-lib/lib/Dimension";
+import Icon from "metabase/components/Icon";
+import IconWrapper from "metabase/components/IconWrapper";
+import { color } from "metabase/lib/colors";
+
+import { Container, Label } from "./DimensionLabel.styled";
+
+DimensionLabel.propTypes = {
+  dimension: PropTypes.instanceOf(Dimension).isRequired,
+};
+
+export default function DimensionLabel({ dimension }) {
+  return (
+    <Container>
+      <IconWrapper borderRadius="4px" bg={color("brand")} p={0}>
+        <Icon name={dimension.icon()} size={12} color={color("white")} />
+      </IconWrapper>
```

Comment:
Should we use a styled component for `IconWrapper` styling?

Reply:
Oh, good idea. I replaced it with a `styled(Icon)`. I ended up tweaking a few things on it, like setting height/width to `1em` so that the size of the label can be driven by `font-size` set on the component.